### PR TITLE
Change up the toggles to move text to the right

### LINKED ISSUE #3908: Change [on|off] toggle to icons with title=on or title=off
screen shot 2017-01-19 at 5 00 22 pm

because while the toggle widget contains translated text, the fixed width breaks on translations

### PR DESCRIPTION
**Pull request by WilliamChiu**
Closes #3908 
 

I'm not too sure what to do about the changing width, or if these changes mess anything else up 😶

### CODE REVIEW

File: lib/css/res.scss [hunk 1]
```diff
@@ -679,8 +679,50 @@ div.usertext-edit {
...
+		-webkit-transform: translateX(30px);
+		-ms-transform: translateX(30px);
+		transform: translateX(30px);
+	}
+
+	&.enabled::after {
+		content: '\00a0On';
+	}
+
+	&:not(.enabled)::after {
```
Comment (andytuba) [inline lib/css/res.scss]: This could probably be `&::after {`

File: lib/utils/createElement.js [hunk 2]
```diff
@@ -20,13 +20,7 @@ export function toggleButton(
 	});
 
 	$('<span>', {
-		class: 'toggleOn noCtrlF',
-		[typeof onText === 'string' ? 'data-text' : 'append']: onText,
```
Comment (andytuba) [inline lib/utils/createElement.js]: What handles custom of/off text now?

File: lib/css/res.scss [hunk 3]
```diff
@@ -679,8 +679,50 @@ div.usertext-edit {
...
+	}
+
+	&.enabled .toggleThumb:before {
+		-webkit-transform: translateX(30px);
+		-ms-transform: translateX(30px);
+		transform: translateX(30px);
+	}
+
+	&.enabled::after {
+		content: '\00a0On';
```
Comment (andytuba) [inline lib/css/res.scss]: Needs overrides for i18n and custom text. Maybe:

In `settingsConsole.js` or something:

```javascript
addCSS(`
    .toggleButton::after { content: '${i18n('toggleDisabledText')}'; }
    .toggleButton.enabled::after { content: '${i18n('toggleEnabledText')}'; }
`);
```

```scss
/* Custom toggle labels */
&[data-enabled-text]::after {
    content: '\00a0' attr(data-enabled-text);
}

&.enabled[data-enabled-text]::after {
    content: '\00a0' attr(data-enabled-text);
}
```

and in `createElement`, set `onText` and `offText` to those data attributes (using `HTMLElement.setAttribute`, not `$.fn.data()`

**Comment by WilliamChiu** [comment]
One thing that bugs me is that the differing widths of 'On' and 'Off' shift nearby content ever so slightly 😢

**Comment by andytuba** [comment]
> differering widths of labels

Maybe we can do something with rendering both labels at once and toggling between them via `visibility`?

**Comment by erikdesjardins** [comment]
Looks great, thanks!

**Comment by WilliamChiu** [comment]
thanks!

**Pull request merged**